teps/0091-trusted-resources.md

```diff
@@ -407,6 +400,67 @@ How does this proposal affect the reusability, simplicity, flexibility
 and conformance of Tekton, as described in [design principles](https://github.com/tektoncd/community/blob/master/design-principles.md)
 -->
 
+### Reusability
+
+<!--
+https://github.com/tektoncd/community/blob/main/design-principles.md#reusability
+
+- Are there existing features related to the proposed features? Were the existing features reused?
+- Is the problem being solved an authoring-time or runtime-concern? Is the proposed feature at the appropriate level
+authoring or runtime?
+-->
+
+There are no existing features in Tekton which can be reused.
+
+This problem is runtime-concern and the proposed feature is at runtime level.
+
+
+### Simplicity
+
+<!--
+https://github.com/tektoncd/community/blob/main/design-principles.md#simplicity
+
+- How does this proposal affect the user experience?
+- What’s the current user experience without the feature and how challenging is it?
+- What will be the user experience with the feature? How would it have changed?
+- Does this proposal contain the bare minimum change needed to solve for the use cases?
+- Are there any implicit behaviors in the proposal? Would users expect these implicit behaviors or would they be
+surprising? Are there security implications for these implicit behaviors?
+-->
+
+This proposal doesn't have any effect if users don't enable it. If enabled, users need to sign the resources and config the public key in cluster to bypass verification, the mutating webhook will be skipped for API resources to avoid the failing verification.
+
+### Flexibility
+
+<!--
+https://github.com/tektoncd/community/blob/main/design-principles.md#flexibility
+
+- Are there dependencies that need to be pulled in for this proposal to work? What support or maintenance would be
+required for these dependencies?
+- Are we coupling two or more Tekton projects in this proposal (e.g. coupling Pipelines to Chains)?
+- Are we coupling Tekton and other projects (e.g. Knative, Sigstore) in this proposal?
+- What is the impact of the coupling to operators e.g. maintenance & end-to-end testing?
+- Are there opinionated choices being made in this proposal? If so, are they necessary and can users extend it with
+their own choices?
+-->
+
+Dependencies need to be pulled into tekton pipelines for this proposal:
+*	github.com/sigstore/sigstore
+
+We are coupling Tekton and Sigstore in this proposal.
+
+What is the impact of the coupling to operators e.g. maintenance & end-to-end testing?
+
+* Need to install sigstore in dogfooding cluster to sign & verify official resources as part of testing
+
+Are there opinionated choices being made in this proposal? If so, are they necessary and can users extend it with
+
+* We choose to use Sigstore to create signer and verifier, users cannot extend/replace it with other options. But users can use whatever keys supported by Sigstore.
```

Comment:
can you explain a bit more about why this required? Is there any Sigstore specific logic or is it just convenient to use the libraries? are there any alternatives?

Reply:
Sure! The reasons in my mind are:

It is convenient to be used as it supports KMS from major cloud providers, and also has the logic to load 
-  https://github.com/sigstore/sigstore/blob/32b357aca9bcd5f75a67667beb775fa13e2b66b4/pkg/signature/verifier.go#L42
- https://github.com/sigstore/sigstore/tree/main/pkg/signature/kms 

It (sigstore/cosign, sigstore/rekor) is also used in Kubernetes, Tekton to sign releases and maintained by Google, Chainguard etc. 

One alternative would be we implement these functions on our own?
Other alternatives would be use libraries such as https://github.com/hashicorp/vault.

Reply:
So it sounds like what we really want form sigstore is the cloud provider KMS interface 🤔  A couple thoughts/questions:

* does sigstore implement these from scratch, or does it pull in libraries that do this? if it pulls in libraries, we could also pull in the same libraries instead of using sigstore
* it sounds like we're not really so much trying to couple ourselves to sigstore as we are trying to use a library that is part of sigstore? if so, maybe we can be explicit that we are only pulling in that library, and NOT the rest of sigstore (e.g. fulcio, rekor, etc.) and that coupling Tekton to those would require further review
* how much work would it be to implement our own?

Reply:
- For sure they pulled in cloud providers' libraries. But I think it's not a simple wrapper of  cloud kms. For reference: https://github.com/sigstore/sigstore/tree/main/pkg/signature/kms 
- Yes we're using github.com/sigstore/sigstore now and not using other libraries. In the TEP I have mentioned that we only pull sigstore/sigstore for now
- It depends on how we want to implement them. Like if just copy paste the code then it shouldn't be of much work. (Not sure if this is a good practice)

Besides, in chains we have already pulled sigstore, cosign, and rekor.
https://github.com/tektoncd/chains/blob/b0c9c54930bce7d8288a3c68447fd4cac1b14176/go.mod#L41-L43

Reply:
> Of interest - we just started some work on the Sigstore side to rewrite a Go client([background](https://docs.google.com/document/d/1aZfk1TlzcuaO0uz76M9D26-gAvoZLn0oCAKvkbuhcPM/edit#heading=h.fw3hnltgfsqv), [repo](https://github.com/sigstore/sigstore-go)). Happy to work together if there's something pipelines needs!

Thanks @wlynch - I cannot access the google doc, do I need to join a specific group to have access?

Reply:
Probably need to join https://groups.google.com/g/sigstore-dev 🙂